lib/cli.ex

```diff
@@ -21,25 +21,32 @@ defmodule Onigumo.CLI do
             {working_dir, switches} = Keyword.pop(switches, :working_dir, File.cwd!())
 
             case switches do
-              [] -> module.main(working_dir)
-              _ -> usage_message()
+              [] ->
+                module.main(working_dir)
+
+              _ ->
+                OptionParser.to_argv(switches)
+                |> then(&"invalid OPTIONS #{Enum.join(&1, ", ")}")
+                |> usage_message()
             end
 
           :error ->
-            usage_message()
+            usage_message("invalid COMPONENT #{component}")
         end
 
-      {_, _, [_ | _]} ->
-        usage_message()
+      {_, _, invalid = [_ | _]} ->
+        Enum.map(invalid, fn {k, _} -> k end)
+        |> then(&"invalid OPTIONS #{Enum.join(&1, ", ")}")
+        |> usage_message()
 
       {_, argv, _} when length(argv) != 1 ->
```

Comment:
This guard is probably unnecessary, I‘ll create an issue for that.

Reply:
OK